### PR TITLE
Group 2 - Solution for Extensible Calculator

### PR DESCRIPTION
**Pull request by confused-binary**
Code results for Group 2

**Team Members**
- Patrick
- Tyler

### CODE REVIEW

File: calculator/main.py [hunk 1]
```diff
@@ -13,10 +13,17 @@ def create_new_calculator(operations=None):
...
     """
-    pass
 
+    calc = {
+            'operations': {},
+            'history': []
+        }
+    if operations:
+        calc['operations'] = operations
+    return calc
```
Comment (martinzugnoni) [inline calculator/main.py:23]: Just to save some lines:

``` python
return {
    'operations': operations or {},
    'history': []
}
```

File: calculator/main.py [hunk 2]
```diff
@@ -56,20 +90,27 @@ def get_history(calc):
...
     """
     Returns the result of the last operation executed in the history.
     """
-    pass
+    
+    # If History key in calc Dictionary has content, rerun last operation.
+    if calc['history']:
+        operation = calc['history'][-1][1]
+        params = calc['history'][-1][2]
+        return calc["operations"][operation](*params)
```
Comment (martinzugnoni) [inline calculator/main.py:114]: As we are storing the result of the operation in the history, it's not necessary to re run the operation. We can just return the result.

**Comment by martinzugnoni** [comment]
Code looks good. You need to be careful with the files you are adding to the revision control. There're many files added in this PR that were not expected as part of the solution. Anyway, you did a great job! ;-)

**Comment by bdauer** [comment]
Looks really good. You used some similar solutions to us (group 4) that Martin pointed out could be written more efficiently or pythonicaly. May be worth looking at our pr for those comments.

File: calculator/main.py [hunk 3]
```diff
@@ -26,7 +33,27 @@ def perform_operation(calc, operation, params):
...
+    if not operation in calc["operations"]:
+        raise(InvalidOperations('{} not supported'.format(operation)))
+    
+    
+    # Pass parameters to lambda stored in operations dictionary.
+    result = calc["operations"][operation](*params)
+    
+    
+    # Obtain date and save to history dictionary.
+    date = datetime.now().strftime('%Y-%m-%d %H:%M:%S')
```
Comment (ghost) [inline calculator/main.py:52]: This seems to be a simpler solution to the code I wrote.

**Comment by ghost** [comment]
You guys did a good job.

**Pull request closed without merging**